### ISSUE TITLE
DX improvement: Make `phenomic start` reuse tabs instead of opening new ones

### ISSUE THREAD

**Issue by DavidWells** [MoOx/phenomic#724]
Taking this from the `create-react-app` CLI.

When you start the `create-react-app` app with `npm start` (if on mac), it checks if your browser has an open tab at the given port and reuses it instead of opening a new tab.

I'd classify this as a nice to have feature for improved developer experience. Keeps browser tabs to a minimum.

[This is how they do it](https://github.com/facebookincubator/create-react-app/blob/master/scripts/start.js#L153-L171) and it runs [this applescript](https://github.com/facebookincubator/create-react-app/blob/master/scripts/utils/chrome.applescript). If not on Mac it defaults back to `opn` package

Implementation should be pretty easy: replace  https://github.com/MoOx/phenomic/blob/161dbc63410e56451af8568377ff4aab38971ac1/src/builder/server.js#L175 with [this](https://github.com/facebookincubator/create-react-app/blob/master/scripts/start.js#L153-L171) and add the applescript to project

The question is: can we use this? I'm not sure what the BSD license means here: https://github.com/facebookincubator/create-react-app/blob/master/scripts/utils/chrome.applescript#L3-L10

What do you think?

**Comment by thangngoc89** [comment]
I'm totally agree with this. But this is for macOS only?

**Comment by DavidWells** [comment]
@thangngoc89 It only works for mac users yes but It will default back to standard `opn` package if the user is on windows etc

**Comment by thangngoc89** [comment]
@DavidWells Could you send a PR. I'll be more than happy to merge it

**Comment by DavidWells** [comment]
@thangngoc89 Do you know about the BSD license stuff? Can we use that applescript as is or do we need to make our own?

**Comment by thangngoc89** [comment]
@DavidWells I don't know about BSD license stuff. But React was licensed with BSD too. We can shoot a tweet to the repo's manager right?

**Comment by DavidWells** [comment]
@thangngoc89 tweeted dan https://twitter.com/DavidWells/status/772638871074906112 waiting for hear back. =)

**Comment by MoOx** [comment]
FYI, they probably ripped this script here http://stackoverflow.com/questions/11735471/apple-script-opening-webpage-on-chrome-and-copy-text (the similarities are more than obvious), so no big deal imo. You can even see that the license header is removed while "ejecting".
Not that this is macOS AND Chrome only.

Could be nice to make it works for other browser :)

**Comment by MoOx** [comment]
I don't see how we can use this script for people that use other browser. This script enforce your to use Chrome right? Maybe I am missing something.
Also, currently, if you navigate in a page (url is not the root anymore) this script does not work since url to open !== url opened.

**Comment by DavidWells** [comment]
The script does this:

```
if (computerHasChromeOpen) {
   // Check for tab in chrome with matching url (localhost:3333)
   if(tabAlreadyOpenReUseIt) {
      // do magic and reuse tab
      return; // stop script
   }
}

do normal opn call here if chrome check fails
```

As for the license from FB. I got this response: https://twitter.com/Vjeux/status/772808842618949633

So does this mean including https://github.com/facebookincubator/create-react-app/blob/master/LICENSE and https://github.com/facebookincubator/create-react-app/blob/master/PATENTS inside the folder that contains this applescript?

An alternative would be adding `create-react-app` as a dev-dependancy but man that would be a lot of extra crap for just this applescript.

**Comment by MoOx** [comment]
Seriously? This code has definitively been mostly ripped from here http://stackoverflow.com/questions/11735471/apple-script-opening-webpage-on-chrome-and-copy-text (2012) or from another place that used another source. This is crazy.

FYI https://twitter.com/MoOx/status/772897845854732289

Anyway, I won't add cra as a dep for a single script (kind of ripped), so I will just go for recoding this using my brain and stack overflow... I guess I will find something to check that chrome is the default browser.... Cause even if I have Chrome open sometimes, it's not my default browser, I don't expect Phenomic to open in Chrome.

But be reassured, we will find a way for this. We could even make a npm package this time, what do you think? What about we create a `open-or-focus {url}`? :D

**Comment by MoOx** [comment]
Btw, if "opn" bother you, you can disable "open" by adding "open: false" in your phenomic config ;)

**Comment by DavidWells** [comment]
Sounds good. It is a convenience feature for sure but is super nice. (seriously try `create-react-app`'s npm start a couple times)

- Refreshes the same url (ie localhost:3333/my-current/path/xyz) instead of loading the homepage everytime (localhost:3333)
- Opens browser back to the same tab and keeps the single tab

The fact that it reopens the same url, not just the homepage, is a time save for me and why I proposed it =)

**Comment by MoOx** [comment]
Closing in favor of #783

**Comment by MoOx** [comment]
Closed by 3917d46529604602bc4645d9e3ffac357c68d015
